src/base/static/components/input-form/index.js

```diff
@@ -289,10 +314,11 @@ class InputForm extends Component {
           onSubmit={this.onSubmit.bind(this)}
         >
           {this.state.fields
+            .slice(this.getStageStartField(), this.getStageEndField())
```

Comment:
I think it would be more clear to set a separate variable `fields` , as well as a `getFieldsFromStage` instead of calling the `getStageField` functions.

ie:

```js
getFieldsFromStage({ fields, stage }) {
  return fields.slice(stage.start_field_index - 1, stage.end_field_index)
}

// ...

const fields = this.selectedCategoryConfig.multi_stage
  ? getFieldsFromStage({
      fields: this.state.fields
      stage:  this.selectedCategoryConfig.multi_stage[this.state.currentStage - 1],
    })
  : this.state.fields

```

What do you think?

Reply:
Ahhh yes. I knew that `.slice(this.getStageStartField(), this.getStageEndField())` looked weird, and I think your suggestion is clearer.